internal/generate/generate.go

```diff
@@ -23,68 +37,79 @@ const (
 )
 
 var (
-	locodeGenerateInPaths        []string
+	locodeGenerateInPaths        stringSliceFlag
 	locodeGenerateSubDivPath     string
 	locodeGenerateAirportsPath   string
 	locodeGenerateCountriesPath  string
 	locodeGenerateContinentsPath string
 	locodeGenerateOutPath        string
+)
 
-	locodeGenerateCmd = &cobra.Command{
-		Use:   "generate",
-		Short: "Generate UN/LOCODE database",
-		Args:  cobra.NoArgs,
-		Run: func(cmd *cobra.Command, _ []string) {
-			locodeDB := csvlocode.New(
-				csvlocode.Prm{
-					Path:       locodeGenerateInPaths[0],
-					SubDivPath: locodeGenerateSubDivPath,
-				},
-				csvlocode.WithExtraPaths(locodeGenerateInPaths[1:]...),
-			)
-
-			airportDB := airportsdb.New(airportsdb.Prm{
-				AirportsPath:  locodeGenerateAirportsPath,
-				CountriesPath: locodeGenerateCountriesPath,
-			})
-
-			continentsDB := continentsdb.New(continentsdb.Prm{
-				Path: locodeGenerateContinentsPath,
-			})
-
-			targetDB := locode.New(locodeGenerateOutPath)
-
-			names := &namesDB{
-				DB:    airportDB,
-				Table: locodeDB,
-			}
-
-			err := locode.FillDatabase(locodeDB, airportDB, continentsDB, names, targetDB)
-			if err != nil {
-				cmd.PrintErrln(err)
-			}
-		},
+func init() {
+	flag.Var(&locodeGenerateInPaths, locodeGenerateInputFlag, "List of paths to UN/LOCODE tables (CSV)")
+	flag.StringVar(&locodeGenerateSubDivPath, locodeGenerateSubDivFlag, "", "Path to UN/LOCODE subdivision database (CSV)")
+	flag.StringVar(&locodeGenerateAirportsPath, locodeGenerateAirportsFlag, "", "Path to OpenFlights airport database (CSV)")
+	flag.StringVar(&locodeGenerateCountriesPath, locodeGenerateCountriesFlag, "", "Path to OpenFlights country database (CSV)")
+	flag.StringVar(&locodeGenerateContinentsPath, locodeGenerateContinentsFlag, "", "Path to continent polygons (GeoJSON)")
+	flag.StringVar(&locodeGenerateOutPath, locodeGenerateOutputFlag, "", "Target path for generated database (directory))")
+}
+
+func Execute() {
```

Comment:
@End-rey now it's a `main` pkg in `internal/` containing executable program which is usually placed in `cmd/`, may be move it there in concert?

Reply:
> may be move it there in concert?

but why? i think it is `internal`ed intentionally. no one is expected to use it, it is for our generations only

Reply:
cuz its a common approach across various Go projects. I can use it from `intenal` as well, and placing program into `cmd` does not force usage

Reply:
@cthulhu-rider Do you propose to make such structure?
```
.
├── cmd
│    └── generate
│            └── main.go
├── internal
└── pkg
```
I agree that then the structure will be better and the command code will be separated from the library code.
But why haven't we used such a structure before? Were there any significant reasons for this?

Reply:
> such structure?

exactly

> But why haven't we used such a structure before?

idk, this isnt a big problem so no one cared i guess

Reply:
That's mostly because this command is completely internal, it's supposed to be used only by package developers. And the only thing properly exposed to the outside world is `pkg/locodedb`.

Reply:
> That's mostly because this command is completely internal, it's supposed to be used only by package developers. And the only thing properly exposed to the outside world is `pkg/locodedb`.

That is, to leave the structure as it is now? `main.go` in `internal`?

Reply:
it's weird to have `main` package in `internal`, at least it should be `internal/generate`. With this, it's easy to make it `cmd/generate` and not to go against the generally accepted approach